functional-tests/test-bed/src/main/java/org/arquillian/smart/testing/ftest/testbed/project/MavenConfigurator.java

```diff
@@ -0,0 +1,117 @@
+
+    private final Model model;
+    private final ProjectConfigurator projectConfigurator;
+
+    MavenConfigurator(Path rootPom, ProjectConfigurator projectConfigurator) {
+        this.model = readPom(rootPom);
+        this.projectConfigurator = projectConfigurator;
+    }
+
+    void addRequiredDependencies() {
+        model.addDependency(smartTestingProviderDependency());
+        final Criteria[] criteria = projectConfigurator.getCriteria();
+        for (final Criteria criterion : criteria) {
+            model.addDependency(criterion.getMavenDependency());
+        }
+    }
+
+    void configureTestRunner() {
+        final Optional<Plugin> testRunnerPluginConfiguration = model.getBuild().getPlugins()
+            .stream()
+            .filter(plugin -> APPLICABLE_PLUGINS.contains(plugin.getArtifactId()))
+            .findFirst();
+
+        if (testRunnerPluginConfiguration.isPresent()) {
+            final Plugin testRunnerPlugin = testRunnerPluginConfiguration.get();
+            testRunnerPlugin.addDependency(smartTestingProviderDependency());
+            final Object configuration = testRunnerPlugin.getConfiguration();
+            if (configuration != null) {
+                final Xpp3Dom configurationDom = (Xpp3Dom) configuration;
+                final Xpp3Dom properties = getOrCreatePropertiesChild(configurationDom);
+                properties.addChild(defineUsageMode());
+                properties.addChild(defineTestSelectionCriteria());
+            }
+        }
+    }
+
+    void update() {
+        try (BufferedWriter pomOut = new BufferedWriter(new FileWriter(model.getPomFile()))) {
+            new MavenXpp3Writer().write(pomOut, model);
+        } catch (IOException e) {
+            throw new RuntimeException("Failed writing updated pom file: " + model.getPomFile().getAbsolutePath(), e);
+        }
+    }
+
+    private Model readPom(Path pom) {
+        final Model model;
+        final File pomFile = pom.toFile();
+        try (BufferedReader pomIn = new BufferedReader(new FileReader(pomFile))) {
+            final MavenXpp3Reader mvnReader = new MavenXpp3Reader();
+            model = mvnReader.read(pomIn);
+            model.setPomFile(pomFile);
+        } catch (IOException | XmlPullParserException e) {
+            throw new RuntimeException("Failed while loading Maven POM from: " + pom.toAbsolutePath(), e);
+        }
+        return model;
+    }
+
+    private Xpp3Dom defineTestSelectionCriteria() {
+        final Xpp3Dom strategies = new Xpp3Dom("strategies");
+        final Criteria[] criteria = projectConfigurator.getCriteria();
+        final StringJoiner stringJoiner = new StringJoiner(",");
+        for (final Criteria criterion : criteria) {
+            stringJoiner.add(criterion.name().toLowerCase());
+        }
+        strategies.setValue(stringJoiner.toString());
+        return strategies;
+    }
+
+    private Xpp3Dom defineUsageMode() {
+        final Xpp3Dom usage = new Xpp3Dom("usage");
```

Comment:
`Maven Extension` - Yeah, this is something that I'd like to hear more about.
I don't know what are you working on exactly and what is your plan/idea with it...